### PR TITLE
Doctest for `PiecewiseLinear` lr scheduler

### PR DESCRIPTION
**Pull request by DevPranjal**
Addresses #2266 

Description: Added doctest for `PiecewiseLinear` lr scheduler

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

**Comment by sdesrozis** [comment]
@DevPranjal Very sorry for the delay... I was so busy :( 

Here is a code that you could use to add some outputs
```python
import torch

from ignite.engine import Events, Engine
from ignite.handlers.param_scheduler import PiecewiseLinear

tensor = torch.zeros([1], requires_grad=True)
optimizer = torch.optim.SGD([tensor], lr=0)

milestones_values = [(1, 1.0), (3, 0.8), (5, 0.2)]
scheduler = PiecewiseLinear(optimizer, "lr", milestones_values=milestones_values)
        
def save_lr(engine):
    lrs.append(optimizer.param_groups[0]["lr"])

trainer = Engine(lambda engine, batch: None)
trainer.add_event_handler(Events.ITERATION_COMPLETED, scheduler)
trainer.add_event_handler(Events.ITERATION_COMPLETED, save_lr)

lrs = []
trainer.run([0] * 6, max_epochs=1)
print(lrs)
# >>> [1.0, 1.0, 0.9, 0.8, 0.5, 0.2]

lrs = []
trainer.run([0], max_epochs=1)
print(lrs)
# >>> [0.2]
```

**Comment by DevPranjal** [comment]
Thanks a lot for this snippet! I am really impressed by the ignite API here, want to learn more such use cases in the future :)

**Comment by sdesrozis** [comment]
> Thanks a lot for this snippet! I am really impressed by the ignite API here, want to learn more such use cases in the future :)

This is the right place to be :)

**Comment by sdesrozis** [comment]
@DevPranjal Are you still working on this ? Thanks !

**Comment by DevPranjal** [comment]
Sorry for the delay... We have our finals coming up at college and hence I have been busy this week. I will make the changes as soon as possible.

**Comment by DevPranjal** [comment]
@sdesrozis, I have added the suggested code, please review. Thanks!

**Comment by sdesrozis** [comment]
Let me a little time to merge a PR I have done on this topic. I have introduced some features we could use in this PR. I keep you in touch.

**Comment by sdesrozis** [comment]
@DevPranjal Could you update this PR accordingly to #2327 ? Thanks !!

**Comment by DevPranjal** [comment]
Yes, I have been following #2327... And now that it is merged, I will update this one by tomorrow :)

File: ignite/handlers/param_scheduler.py [hunk 1]
```diff
@@ -987,6 +987,78 @@ class PiecewiseLinear(ParamScheduler):
         # from 0.3 to 0.1 between 21st and 30th iterations and remains 0.1 until the end of the iterations.
         #
```
Comment (DevPranjal) [inline ignite/handlers/param_scheduler.py:988]: Should I remove this snippet? @sdesrozis

**Comment by DevPranjal** [comment]
https://github.com/pytorch/ignite/blob/60e2af4b4bd85bb6e6d6bdda3a5203fbaf3140f1/ignite/handlers/param_scheduler.py#L978-L988

Should I remove this snippet? @sdesrozis 

Also, sorry for the mess in commit history

**Comment by sdesrozis** [comment]
Your test should reflect the current test, even adding a few comments to be very explicit about what is done.

**Comment by DevPranjal** [comment]
@sdesrozis Gentle ping, in case you missed the last commit. I have added the comments as suggested

**Review by sdesrozis — APPROVED** [review]
@DevPranjal Thank you ! LGTM !
